learn/core_concepts/documents.md

```diff
@@ -1,26 +1,58 @@
-- **Value**: the second part of a field, consisting of data of any valid JSON type
-- **[Primary Field][primary-field]**: A special field that is mandatory in all documents. It contains the primary key and document identifier.
-- **[Primary Key][primary-key]**: the attribute of the primary field. **All documents in the same index must possess the same primary key.** Its associated value is the document identifier.
-- **[Document Identifier][document-id]**: the value of the primary field. **Every document in a given index must have a unique identifier**.
+- **Document**: An object which contains data in the form of one or more fields
+- **[Field](#fields)**: A set of two data items that are linked together: an attribute and a value
+- **Attribute**: The first part of a field. Acts as a name or description for its associated value
+- **Value**: The second part of a field, consisting of data of any valid JSON type
+- **[Primary Field](#primary-field)**: A special field that is mandatory in all documents. It contains the primary key and document identifier
+
+## Fields
+
+A field is a set of two data items linked together: an attribute and a value. Documents are made up of fields.
+
+An attribute functions a bit like a variable in most programming languages, i.e., it is a name that allows you to store, access, and describe some data. That data is the attribute's value. In the case of strings, a value **[can contain at most 65535 positions](/learn/advanced/known_limitations.md#maximum-number-of-words-per-attribute)**. Words exceeding the 65535 position limit will be ignored.
+
+Every field has a data type dictated by its value. Every value must be a valid [JSON data type](https://www.w3schools.com/js/js_json_datatypes.asp).
+
+If a field contains an object, Meilisearch flattens it during indexation using dot notation and brings the object's keys and values to the root level of the document itself. This flattened object is an intermediary representation and you will get the original structure upon search. You can read more about this in our [dedicated guide](/learn/advanced/datatypes.md#objects).
+
+You can apply [ranking rules](/learn/core_concepts/relevancy.md#ranking-rules) to some fields. For example, you may decide recent movies should be more relevant than older ones. You can also configure how Meilisearch handles certain fields at an [index level](/learn/configuration/settings.md) in the settings.
+
+### Displayed and searchable fields
+
+By default, all fields in a document are both displayed and searchable. You can modify this behavior using the [update settings endpoint](/reference/api/settings.md#update-settings), or the respective update endpoints for [displayed attributes](/reference/api/displayed_attributes.md#update-displayed-attributes), and [searchable attributes](/reference/api/searchable_attributes.md#update-searchable-attributes) so that a field is:
```

Comment:
Honest question: is it clear enough what we mean by displayed and searchable fields? Would users benefit from a sentence here defining these two "types" of fields?